templates/templates.go

```diff
@@ -140,15 +143,92 @@ func BulkBuildInfo(w io.Writer, b *ddao.Build) {
 }
 
 func PkgInfo(w io.Writer, res ddao.GetSingleResultRow) {
+	// Define what to fetch - these represent the stages of the build process
+	// and are used to fetch the corresponding log files.
+	stages := []string{
+		"work.log",
+		"pre-clean.log",
+		"checksum.log",
+		"depends.log",
+		"configure.log",
+		"build.log",
+		"install.log",
+	}
+
+	// Create results slice
+	results := make([]URLRequestResult, len(stages))
+
+	// Create a WaitGroup to wait for all goroutines
+	var wg sync.WaitGroup
+	wg.Add(len(stages))
+
+	// Create a context with a 10-second timeout
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// Fetch URLs in parallel
+	for i, s := range stages {
+		// Construct the URL for the current stage
+		url := fmt.Sprintf("%s%s/%s", res.BaseURL(), res.PkgName, s)
+
+		// Start a goroutine for each URL to fetch stage data
+		go func(index int, stage string, reqURL string) {
+			defer wg.Done()
+
+			// Initialize result with ID and URL
+			results[index] = URLRequestResult{
+				ID:  stage,
+				URL: reqURL,
+			}
+
+			// Create a new request with the context
+			req, err := http.NewRequestWithContext(ctx, "GET", reqURL, nil)
+			if err != nil {
+				results[index].StatusCode = http.StatusInternalServerError
+				results[index].Error = err.Error()
+				return
+			}
+
+			// Make the request
+			resp, err := http.DefaultClient.Do(req)
+			if err != nil {
+				results[index].StatusCode = http.StatusInternalServerError
+				results[index].Error = err.Error()
+				return
+			}
+			defer resp.Body.Close()
+
+			// Read the response body
+			body, err := io.ReadAll(resp.Body)
+			if err != nil {
+				results[index].StatusCode = resp.StatusCode
+				results[index].Error = err.Error()
+				return
+			}
+
+			// Store the result
+			results[index].StatusCode = resp.StatusCode
+			results[index].Data = string(body)
+		}(i, s, url) // i is index, s is stage, url is the constructed URL
+	}
+
+	// Wait for all requests to complete
+	wg.Wait()
+
+	// Create the structure to pass to the template
 	s := struct {
-		Res *ddao.GetSingleResultRow
+		Res     *ddao.GetSingleResultRow
+		URLData []URLRequestResult
```

Comment:
Doesn't need to be in this PR but: what do you think about a cache for `URLData` with a relatively short duration?